test/framework/easyblock.py

```diff
@@ -2172,46 +2172,96 @@ def test_extension_set_start_dir(self):
         cwd = os.getcwd()
         self.assertTrue(os.path.exists(cwd))
 
-        def check_ext_start_dir(expected_start_dir):
+        def check_ext_start_dir(expected_start_dir, unpack_src=True):
             """Check start dir."""
             # make sure we're in an existing directory at the start
             change_dir(cwd)
             eb = EasyBlock(ec['ec'])
             eb.extensions_step(fetch=True, install=False)
             # extract sources of the extension
             ext = eb.ext_instances[-1]
-            ext.run(unpack_src=True)
-            abs_expected_start_dir = os.path.join(eb.builddir, expected_start_dir)
-            self.assertTrue(os.path.samefile(ext.cfg['start_dir'], abs_expected_start_dir))
-            self.assertTrue(os.path.samefile(os.getcwd(), abs_expected_start_dir))
+            ext.run(unpack_src=unpack_src)
+
+            if expected_start_dir is None:
+                self.assertIsNone(ext.start_dir)
+            else:
+                self.assertTrue(os.path.isabs(ext.start_dir))
+                if ext.start_dir != os.sep:
+                    self.assertFalse(ext.start_dir.endswith(os.sep))
+                if os.path.isabs(expected_start_dir):
+                    abs_expected_start_dir = expected_start_dir
+                else:
+                    abs_expected_start_dir = os.path.join(eb.builddir, expected_start_dir)
+                self.assertEqual(ext.start_dir, abs_expected_start_dir)
+                if not os.path.exists(eb.builddir):
+                    eb.make_builddir()  # Required to exist for samefile
+                self.assertTrue(os.path.samefile(ext.start_dir, abs_expected_start_dir))
+            if unpack_src:
+                self.assertTrue(os.path.samefile(os.getcwd(), abs_expected_start_dir))
```

Comment:
`abs_expected_start_dir` might not be set if `expected_start_dir` is None

Reply:
Yeah, I guess in that case this line would raise a TypeError. That is fine as it shouldn't ever happen: If we unpack then start_dir and hence `expected_start_dir` must not be `None` so an earlier assertion would already be triggered.

Reply:
it's true that several other assertions will fail in such a case :+1: